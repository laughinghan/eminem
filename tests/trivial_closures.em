let .sum3 ^a b c (a.plus b.plus c)
print "(sum3 1 2 3): " (sum3 1 2 3) ; println ""

let .avg2 ^x ^y ((x.plus y).divide 2)
print "(avg2 10 11): " (avg2 10 11) ; println ""

let .add10 (sum3 5 5)
print "(add10 11): " (add10 11) ; println ""

^x (println x) "something"
