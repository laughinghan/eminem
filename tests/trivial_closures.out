(sum3 1 2 3): 6
(avg2 10 11): 10.5
(add10 11): 21
something
